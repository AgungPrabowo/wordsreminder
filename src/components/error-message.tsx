import React from 'react'
import { Text } from '@components/text'
import { ERROR_MESSAGE } from '@e2e/ids'

type Props = {
  message: string
  children?: never
}

const ErrorMessage = ({ message }: Props) => (
  <Text testID={ERROR_MESSAGE} color="danger" fontSize={20}>
    {message}
  </Text>
)

export { ErrorMessage }
