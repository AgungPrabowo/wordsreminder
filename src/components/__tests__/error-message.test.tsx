import React from 'react'
import { render, shallow } from 'react-native-testing-library'
import { ErrorMessage } from '../error-message'

describe('ErrorMessage', () => {
  const message = 'A message'
  it('should render', async () => {
    const { queryByText } = render(<ErrorMessage message={message} />)

    expect(queryByText(message)).not.toBeNull()
  })

  it('should match snapshot', () => {
    const { output } = shallow(<ErrorMessage message={message} />)
    expect(output).toMatchSnapshot()
  })
})
