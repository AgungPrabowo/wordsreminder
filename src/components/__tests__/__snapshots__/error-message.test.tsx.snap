// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should match snapshot 1`] = `
<Text
  color="danger"
  fontSize={20}
  testID="error-message"
>
  A message
</Text>
`;
